### PR TITLE
PCHR-2184: Page reloads when manager creates request for staff

### PR DESCRIPTION
**Pull request by dhavalsharma**
## Overview
Page reloads when manager creates request on behalf of staff.

## After
manager-actions

## Technical Details
Added event to listen when new request is created to refresh page.

## Comments
NA

---

- [X ] Tests Pass

## Bug 2
The left status list was not getting updated after change to leave requests. Now it does not get data from cache when clicked on left statuses menu.

## GIF
manager-actions

### CODE REVIEW

File: uk.co.compucorp.civicrm.hrleaveandabsences/js/angular/src/leave-absences/manager-leave/components/manager-leave-requests.js [hunk 1]
```diff
@@ -490,6 +487,21 @@ define([
       }
     }
 
+    /**
+     * Register events which will be called by other modules
+     */
+    function registerEvents() {
+      $rootScope.$on('LeaveRequest::updatedByManager', function () {
```
Comment (deb1990) [inline uk.co.compucorp.civicrm.hrleaveandabsences/js/angular/src/leave-absences/manager-leave/components/manager-leave-requests.js]: Same listener is present for both events, create private function.
```
$rootScope.$on('LeaveRequest::updatedByManager', listener);
$rootScope.$on('LeaveRequest:: new', listener);

function listener() {
  leaveRequestAPICache = false;
  vm.refresh();
}

File: uk.co.compucorp.civicrm.hrleaveandabsences/js/angular/src/leave-absences/manager-leave/components/manager-leave-requests.js [hunk 2]
```diff
@@ -491,15 +497,20 @@ define([
...
+    }
 
-      $rootScope.$on('LeaveRequest::new', function () {
-        leaveRequestAPICache = false;
-        vm.refresh();
-      });
+    /**
+     * Refresh leave requests from server
+     */
+    function refreshFromServer() {
```
Comment (deb1990) [inline uk.co.compucorp.civicrm.hrleaveandabsences/js/angular/src/leave-absences/manager-leave/components/manager-leave-requests.js]: Please keep alphabetical order

File: uk.co.compucorp.civicrm.hrleaveandabsences/js/angular/src/leave-absences/manager-leave/components/manager-leave-requests.js [hunk 3]
```diff
@@ -236,7 +241,7 @@ define([
      */
     vm.refreshWithFilter = function (status) {
       vm.filters.leaveRequest.leaveStatus = status;
-      vm.refresh();
+      refreshFromServer();
```
Comment (deb1990) [inline uk.co.compucorp.civicrm.hrleaveandabsences/js/angular/src/leave-absences/manager-leave/components/manager-leave-requests.js]: Why this is changed to cache false?

File: uk.co.compucorp.civicrm.hrleaveandabsences/js/angular/src/leave-absences/manager-leave/components/manager-leave-requests.js (hunk 2 above)
Comment (deb1990) [inline uk.co.compucorp.civicrm.hrleaveandabsences/js/angular/src/leave-absences/manager-leave/components/manager-leave-requests.js]: `refreshFromServer` is confusing as refresh should happen always from server. Perhaps a better name like `refreshWithoutCache` is more meaningful.

File: uk.co.compucorp.civicrm.hrleaveandabsences/js/angular/src/leave-absences/manager-leave/components/manager-leave-requests.js (hunk 3 above)
Comment (dhavalsharma) [inline uk.co.compucorp.civicrm.hrleaveandabsences/js/angular/src/leave-absences/manager-leave/components/manager-leave-requests.js]: for the menu refresh fix as it was loading older cached data

**Pull request merged**